### DISCUSSION TITLE
How to extract Principal Components as numpy arrays?

### DISCUSSION THREAD

**Discussion by kcnguyen3191** [GAA-UAM/scikit-fda#637] [Q&A] [answered]
Hi All,

I have been trying to see if I can extract the time series of the principal components as numpy arrays of time series. I know `components_` will give me the time series, but they are in `FDataGrid`. I tried to use `.to_numpy()` function in this [document](https://fda.readthedocs.io/en/latest/modules/autosummary/skfda.representation.grid.FDataGrid.html#skfda.representation.grid.FDataGrid.to_numpy), but it just seems to wrap numpy array around `FDataGrid` data type variable. Is there a way to this?

Thanks!

**Comment by vnmabus** [comment, marked as answer]
You can find the array containing the values of the function in the `data_matrix` attribute of an `FDataGrid`, as explained [here](https://fda.readthedocs.io/en/latest/auto_tutorial/plot_getting_data.html#the-fdatagrid-class). Maybe that is what you want?

**Reply by kcnguyen3191** [reply]
Yes, this is exactly what I want. Thank you!
